src/test/scala/nl.knaw.dans.easy.multideposit/SplitMultiDepositAppSpec.scala

```diff
@@ -277,10 +284,30 @@ class SplitMultiDepositAppSpec extends TestSupportFixture with MockFactory with
       }
     }
 
-    "ruimtereis01" should behave like bagContents("ruimtereis01", expectedDataContentRuimtereis01)
-    "ruimtereis02" should behave like bagContents("ruimtereis02", expectedDataContentRuimtereis02)
-    "ruimtereis03" should behave like bagContents("ruimtereis03", expectedDataContentRuimtereis03)
-    "ruimtereis04" should behave like bagContents("ruimtereis04", expectedDataContentRuimtereis04)
+    it should "check report.csv" in {
+      reportFile.toJava should exist
+
+      val header :: lines = reportFile.lines.toList
+      val reportLines = lines.collect {
+        case s if s startsWith ruimtereis01 => ruimtereis01 -> s
+        case s if s startsWith ruimtereis02 => ruimtereis02 -> s
+        case s if s startsWith ruimtereis03 => ruimtereis03 -> s
+        case s if s startsWith ruimtereis04 => ruimtereis04 -> s
+      }.toMap
+      // taken from https://stackoverflow.com/a/6640851/2389405
+      val uuidRegex = "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"
+
+      header shouldBe "DATASET,UUID,BASE-REVISION"
+      reportLines(ruimtereis01) should fullyMatch regex s"^$ruimtereis01,$uuidRegex,d5e8f0fb-c374-86eb-918c-b06dd5ae5e71$$"
+      reportLines(ruimtereis02) should fullyMatch regex s"^$ruimtereis02,($uuidRegex),\\1$$"
+      reportLines(ruimtereis03) should fullyMatch regex s"^$ruimtereis03,($uuidRegex),\\1$$"
+      reportLines(ruimtereis04) should fullyMatch regex s"^$ruimtereis04,$uuidRegex,773dc53a-1cdb-47c4-992a-254a59b98376$$"
+    }
+
+    ruimtereis01 should behave like bagContents(ruimtereis01, expectedDataContentRuimtereis01)
+    ruimtereis02 should behave like bagContents(ruimtereis02, expectedDataContentRuimtereis02)
+    ruimtereis03 should behave like bagContents(ruimtereis03, expectedDataContentRuimtereis03)
+    ruimtereis04 should behave like bagContents(ruimtereis04, expectedDataContentRuimtereis04)
```

Comment:
'behave like' is a new concept for me.

Reply:
http://www.scalatest.org/user_guide/sharing_tests

It's basically a way to write your tests once and call them for different circumstances.

As you see in this class, `"xxx" should behave like f` defines a function `f` with the actual tests in it. In this use case we also provide arguments to the function to adjust the behaviour of the function according to the specific test case.

I found it to be quite a useful tool; perhaps you can think of some use cases in your code as well...

Reply:
Thanks, indeed 👍  a solution to a current problem as I'm developing twin tests, one to check against expected output, one to validate against a scheme. Can't have the two in a single test with two `should`s as as the validation would be ignored when running offline.

Reply:
nice, then those two tests would be inside a function and you would call that function in the `should behave like ...`
curious to see the results!